### DISCUSSION TITLE
unexpected internet requests when file is loaded

### DISCUSSION THREAD

**Discussion by atomGit** [Zettlr/Zettlr#4582] [Q&A]
i'm seeing unexpected network requests when opening an md file with links in it - it appears each link is being verified or something - confirmed with wireshark on linux

update functions are disabled

**Comment by nathanlesage** [comment]
> i'm seeing unexpected network requests when opening an md file with links in it - it appears each link is being verified or something

They're not verified — Zettlr fetches preview information, basically the same that happens on every other website as well. You can turn this off in the settings:

![image](https://github.com/Zettlr/Zettlr/assets/17251683/d699502a-ca45-468d-b3d5-7d7822798fb1)

**Comment by atomGit** [comment]
i don't have zettlr installed at the moment (frankly i'm a bit put off by the electron dependency) but i assume with link previews disabled, "[this](http://example.com)" turns into `[this](http://example.com)`

i'm curious about the rendering - i see in the code...

```
import {
  BrowserWindow,
  type BrowserWindowConstructorOptions
} from 'electron'
```

how is electron rendering content? is it using chrome/chromium, gecko/quantum or some other engine? and does the user have any control over it (config files)?

**Comment by nathanlesage** [comment]
> i don't have zettlr installed at the moment (frankly i'm a bit put off by the electron dependency) but i assume with link previews disabled, "[this](http://example.com/)" turns into `[this](http://example.com)`

No, and I don't think what would make you believe that. Removing formatting characters to make a link display without the lengthy link target does not require an internet connection. Link previews is literally what I described: Fetching some information from the link to give you a preview of what the site is about, e.g., fetching the abstract of a Wikipedia article, and the image. Look on Twitter, Telegram, Reddit, etc. – all of these do the same (albeit their previews look differently).

> how is electron rendering content? is it using chrome/chromium, gecko/quantum or some other engine? and does the user have any control over it (config files)?

As you described: It's nothing more than Chrome's rendering engine with an attached NodeJS server. I don't understand what exactly you mean with "control over it" — control over what?
